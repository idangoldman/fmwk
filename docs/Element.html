<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Element</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Element</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Element<span class="signature">(selector)</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>Wrapper class for DOM elements, initiate by passing a selector.</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Element"><span class="type-signature"></span>new Element<span class="signature">(selector)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Creates an element instance of the query selector passed in.</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>selector</code></td>
            

            <td class="type">
            
                
<span class="param-type">String</span>


            
            </td>

            

            

            <td class="description last"><p>DOM query selector</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="element_index.js.html">element/index.js</a>, <a href="element_index.js.html#line11">line 11</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Throws:</h5>
<ul>
        <li>

    <div class="param-desc">
    
        <p>When no query selector passed.</p>
    
    </div>

</li>
    
        <li>

    <div class="param-desc">
    
        <p>When no HTML elements were found via passed query selector.</p>
    
    </div>

</li>
    </ul>







    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="element"><span class="type-signature">(private) </span>element<span class="type-signature"> :<a href="global.html#HTMLElement">HTMLElement</a></span></h4>




<div class="description">
    <p>An instance of the DOM element.</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type"><a href="global.html#HTMLElement">HTMLElement</a></span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="element_index.js.html">element/index.js</a>, <a href="element_index.js.html#line17">line 17</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="raw"><span class="type-signature"></span>raw<span class="type-signature"></span></h4>




<div class="description">
    <p>Get the instance of DOM element.</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="element_index.js.html">element/index.js</a>, <a href="element_index.js.html#line41">line 41</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="selector"><span class="type-signature">(private) </span>selector<span class="type-signature"> :<a href="global.html#QuerySelector">QuerySelector</a></span></h4>




<div class="description">
    <p>An instance of the query selector string.</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type"><a href="global.html#QuerySelector">QuerySelector</a></span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="element_index.js.html">element/index.js</a>, <a href="element_index.js.html#line24">line 24</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="toString"><span class="type-signature"></span>toString<span class="type-signature"></span></h4>




<div class="description">
    <p>Get the query selector string.</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="element_index.js.html">element/index.js</a>, <a href="element_index.js.html#line49">line 49</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Element.html">Element</a></li></ul><h3>Global</h3><ul><li><a href="global.html#eventNamesValidation">eventNamesValidation</a></li><li><a href="global.html#findReverseBranch">findReverseBranch</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Thu Sep 05 2019 22:47:20 GMT+0300 (Israel Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>