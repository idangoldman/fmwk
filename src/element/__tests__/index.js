import Element from '../index';

describe('Element class tested', () => {
  beforeAll(() => {
    document.body.innerHTML = `
      <a href="#" class="link">Link</a>
    `;
  });

  test('Should be an instance of HTMLElement', () => {
    const element = new Element('.link');
    expect(element.raw).toBeInstanceOf(HTMLElement);
  });

  test('Should throw an error, no selectors were passed', () => {
    const element = () => new Element();
    expect(element).toThrowErrorMatchingSnapshot();
  });

  test('Should throw an error, selector not found', () => {
    const element = () => new Element('.linked');
    expect(element).toThrowErrorMatchingSnapshot();
  });
});
