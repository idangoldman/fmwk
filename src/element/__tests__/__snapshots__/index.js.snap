// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element class tested Should throw an error, no selectors were passed 1`] = `"- No query selector passed."`;

exports[`Element class tested Should throw an error, selector not found 1`] = `"- No HTML element was found with '.linked' selector."`;
